Scrum 24 fe page view single post

AidaBur opened the pull request:
Single Post View (modal + full page)

- When clicking a post in the list, it opens as a route-based modal (/app/posts/:id), and the background list remains visible as a dimmed overlay
- The modal has its own URL and can be shared
- If someone opens a post directly by link, it loads as a separate full page (PostPage.jsx)
- On PostPage we show the full layout: image, details, giver info, and related posts from the same category
- Related posts open in their own page (not a modal)
- Added a breadcrumb link to allow the user to return back to the post list
* Left space on the left side for the sidebar in the layout
<img width="1792" alt="Screen Shot 2025-04-14 at 7 32 33 PM" src="https://github.com/user-attachments/assets/287fdb38-bb59-4baa-b39b-1004adde8576" />
<img width="1792" alt="Screen Shot 2025-04-14 at 7 40 17 PM" src="https://github.com/user-attachments/assets/3c7e707a-dca2-4ad0-9cf8-d28980d2045c" />

mateuspiresl commented on src/pages/Post.jsx:
Since this component is not a page, it would be better if located in the `/src/components` folder.

mateuspiresl commented on src/pages/PostPage.jsx:
```diff
@@ -0,0 +1,151 @@
+
+            <div className="w-full lg:w-1/2 mt-8 lg:mt-0">
+              <h2 className="text-4xl font-extrabold font-montserrat text-primary mb-2">
+                {currentPost.title}
+              </h2>
+              <p className="text-base text-gray mb-6">{currentPost.category}</p>
+
+              <div className="mb-6">
+                <h3 className="text-lg font-semibold">Details:</h3>
+                <p>{currentPost.description}</p>
+              </div>
+
+              <div className="mb-6">
+                <h3 className="text-lg font-semibold">Address to meet:</h3>
+                <p>{currentPost.location}</p>
+                <div className="w-full h-52 bg-gray-light rounded-xl flex items-center justify-center text-gray text-sm mt-4">
+                  Map will be displayed here
+                </div>
+              </div>
+
+              <div className="mb-6">
+                <h3 className="text-lg font-semibold mb-2">
+                  Giver Information:
+                </h3>
+                <div className="flex items-center gap-3 mb-2">
+                  <img
+                    src={currentPost.user?.avatar || "/icons/avatar.svg"}
+                    alt={currentPost.user?.name || "User avatar"}
+                    className="w-5 h-5 rounded-full object-cover"
+                  />
+                  <span>{currentPost.user?.name || "Unknown user"}</span>
+                </div>
+
+                <div className="flex items-center gap-2">
+                  <img src="/icons/email.svg" alt="Email" className="w-5 h-5" />
+                  <span>{currentPost.user?.email || "No email provided"}</span>
+                </div>
+              </div>
+            </div>
+          </div>
```
Seems that this entire area is the `Post` component - you could you it here.

```suggestion
          <Post />
```

AidaBur replied:
I moved that part into a separate `Post` component and now use it in both `PostPage` and `PostModal`. It makes everything feel way cleaner and easier to manage. Appreciate the tip!

mateuspiresl commented on src/pages/PostModal.jsx:
This one could also be better located outside of the pages folder.

mateuspiresl reviewed (changes requested):
I didn't know about this pattern for the routes - nice addition.

mateuspiresl commented on src/pages/Post.jsx:
```diff
@@ -1,6 +1,114 @@
-  return <h1>Item Detail Page — ID: {id}</h1>;
+  const navigate = useNavigate();
+  const { getPost, currentPost, isLoading, error } = usePosts();
+  const [selectedPhoto, setSelectedPhoto] = useState(null);
+
+  useEffect(() => {
+    getPost(Number(id));
+  }, [id, getPost]);
+
+  useEffect(() => {
+    if (currentPost?.photos?.length > 0) {
+      setSelectedPhoto(currentPost.photos[0]);
+    }
+  }, [currentPost]);
```
Alternative implementation for the sake of exercising the solution - this is just a suggestion.

You can drop the second `useEffect` if you use the state to store the index instead of the actual value: 

```suggestion
  const [selectedPhotoIndex, setSelectedPhotoIndex] = useState(0);
  const selectedPhoto = currentPost?.photos?.[selectedPhotoIndex] || null;

  useEffect(() => {
    getPost(Number(id));
  }, [id, getPost]);
```

It is always good to not introduce a new hook if you can. Every hook added has a potential performance impact.

AidaBur commented:
- Removed unused props in App.jsx
- Moved `PostModal` and `Post` to the components folder
- Extracted shared post UI into a reusable `Post` component used by both `PostPage` and `PostModal`
- Replaced selectedPhoto state with selectedPhotoIndex to simplify logic and remove an extra useEffect

mateuspiresl commented on src/components/PostModal.jsx:
```diff
@@ -0,0 +1,50 @@
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const { getPost, currentPost, isLoading, error } = usePosts();
+  const [selectedPhotoIndex, setSelectedPhotoIndex] = useState(0);
+  const selectedPhoto = currentPost?.photos?.[selectedPhotoIndex] || null;
```
One last suggestion: Move this part to the `Post` component. It has access to the post data, so it can certainly handle the images.

AidaBur replied:
Thanks! Moved the photo selection logic into the Post component as suggested

mateuspiresl reviewed (approved):
Good to go 💯
